src/main/java/org/jitsi/jigasi/transcription/Transcriber.java

```diff
@@ -0,0 +1,550 @@
+    {
+        return State.TRANSCRIBING.equals(this.state);
+    }
+
+    /**
+     * Get whether the transcribed has been stopped and will not have any new
+     * results coming in. This is always true after every
+     * {@link TranscriptionListener} has has their
+     * {@link TranscriptionListener#completed()} method called
+     *
+     * @return true when the transcribed has stopped and no new results will
+     * ever come in
+     */
+    public boolean finished()
+    {
+        return State.FINISHED.equals(this.state);
+    }
+
+    /**
+     * Get whether the transcribed has been stopped and is processing the
+     * last audio fragments before it will be finished
+     *
+     * @return true when the transcriber is waiting for the last results to come
+     * in, false otherwise
+     */
+    public boolean finshingUp()
+    {
+        return State.FINISHING_UP.equals(this.state);
+    }
+
+    /**
+     * Provides the (ongoing) transcription of the conference this object
+     * is transcribing
+     *
+     * @return the Transcription object which will be updated
+     * as long as this object keeps transcribing
+     */
+    public Transcription getTranscription()
+    {
+        return transcription;
+    }
+
+    /**
+     * Add a TranscriptionListener which will be notified when the Transcription
+     * is updated due to new TranscriptionResult's coming in
+     *
+     * @param listener the listener which will be notified
+     */
+    public void addTranscriptionListener(TranscriptionListener listener)
+    {
+        listeners.add(listener);
+    }
+
+    /**
+     * Remove a TranscriptionListener such that it will no longer be
+     * notified of new results
+     *
+     * @param listener the listener to remove
+     */
+    public void removeTranscriptionListener(TranscriptionListener listener)
+    {
+        listeners.remove(listener);
+    }
+
+    /**
+     * The transcriber can be used as a RawStreamListener to listen for new
+     * audio packets coming in through a MediaDevice. It will try to filter
+     * them based on the SSRC of the packet. If the SSRC does not match
+     * a participant added to the transcribed, an exception will be thrown
+     *
+     * An ExecutorService is used to offload work on the thread managing
+     * audio packets
+     *
+     * @param receiveStream the stream from which the audio was received
+     * @param buffer the containing the audio as well as meta-data
+     */
+    @Override
+    public void samplesRead(ReceiveStream receiveStream, Buffer buffer)
+    {
+        if(!isTranscribing())
+        {
+            return;
+        }
+
+        executorService.submit(() ->
+        {
+            // ReceiveStream uses unsigned longs
+            long ssrc = receiveStream.getSSRC() & 0xffffffffL;
+
+            Participant p = participants.get(ssrc);
+            if(p != null)
+            {
+                participants.get(ssrc).giveBuffer(buffer);
```

Comment:
> Why call get() a second time when you already have the result in "p"?
fixed

> I think the code which makes the copy should run in the mixer thread
 and it should submit the task only when it needs to send send a request (this will also have less tasks to submit, since not every Buffer results in a request being sent)

Okay, I've moved the local buffering to the mixer thread.